src/game_battlealgorithm.cpp

```diff
@@ -78,13 +78,13 @@ static inline int ToHitPhysical(Game_Battler *source, Game_Battler *target, int
 	return to_hit;
 }
 
-static void BattlePhysicalStateHeal(int physical_rate, std::vector<int16_t>& target_states, std::vector<Game_BattleAlgorithm::StateEffect>& states) {
+static void BattlePhysicalStateHeal(int physical_rate, std::vector<int16_t>& target_states, const PermanentStates& ps, std::vector<Game_BattleAlgorithm::StateEffect>& states) {
```

Comment:
>  Add PermanentStates concept 

(Flagging the whole commit). Is there any sanity check for PermanentStates? What happens when a weapon inflicts a state that is out-of-bounds?
Looks filtered in next commit "Support equipment inflicting states "?

Reply:
The warnings will show up when the equipment tries to add the states.

I added a new commit to prevent adding these invalid states to the `PermanentStates` object.